.travis/runflake8.sh

```diff
@@ -10,6 +10,8 @@
 #
 #   RUN_FLAKE8_DISABLED
 #     if set to an arbitrary non-empty value, flake8 will be not executed
+#   RUN_FLAKE8_IGNORE
```

Comment:
The black wrapper uses a generic "EXTRA_ARGS" variable and this a specific "IGNORE" variable. It seems using "RUN_FLAKE8_EXTRA_ARGS" here would make the template more consistent and provide a similar benefit.

Reply:
do you mean there should be simply `RUN_FLAKE8_EXTRA_ARGS` which would allow users to provide `--ignore value,list` to runflake8.sh, and get rid of `RUN_FLAKE8_IGNORE`?

Reply:
Or, add a `RUN_FLAKE8_EXTRA_ARGS` in addition to `RUN_FLAKE8_IGNORE`, similar to how black has special purpose `RUN_BLACK_INCLUDE` and `RUN_BLACK_EXCLUDE` (because they are more commonly used and cover the vast majority of the cases where the user wants to customize the behavior?) and `RUN_BLACK_EXTRA_ARGS` for everything else?

Reply:
I'm assuming it is the latter

Reply:
I guess I would also remvoe `RUN_BLACK_IN/EXCLUDE` and only allow enabling/disabling and extra args. But it seems the challenge with black is that it needs non-empty default values for include/exclude, therefore the specific variables are needed.

Reply:
> needs non-empty default values for include/exclude

needs in what sense?  That they are required?  They don't appear to be mandatory arguments:
```
Usage: black [OPTIONS] [SRC]...

  The uncompromising code formatter.

Options:
...
  --include TEXT                  A regular expression that matches files and
                                  directories that should be included on
                                  recursive searches.  An empty value means
                                  all files are included regardless of the
                                  name.  Use forward slashes for directories
                                  on all platforms (Windows, too).  Exclusions
                                  are calculated first, inclusions later.
                                  [default: \.pyi?$]

  --exclude TEXT                  A regular expression that matches files and
                                  directories that should be excluded on
                                  recursive searches.  An empty value means no
                                  paths are excluded. Use forward slashes for
                                  directories on all platforms (Windows, too).
                                  Exclusions are calculated first, inclusions
                                  later.  [default: /(\.eggs|\.git|\.hg|\.mypy
                                  _cache|\.nox|\.tox|\.venv|\.svn|_build|buck-
                                  out|build|dist)/]
```

Reply:
The template uses non-empty default values:

```
DEFAULT_INCLUDE='^[^.].*\.py$'
DEFAULT_EXCLUDE='/(\.[^.].*|tests/roles)/'
```
From the value of the exclude I assume this is because `/tests/roles` might contain a symlink back to `/` causing an infinite loop.

Reply:
So they weren't really mandatory, but they were convenient, or were otherwise useful to have as separate environment variables, since they could have been implemented using `RUN_BLACK_EXTRA_ARGS`

So for the purpose of setting a good coding style precedent for this and future code (as opposed to being consistent with runblack.sh), I should get rid of `RUN_FLAKE8_IGNORE` and instead allow this and other overrides to be done solely via `RUN_FLAKE8_EXTRA_ARGS`?

Reply:
> So they weren't really mandatory, but they were convenient, or were otherwise useful to have as separate environment variables, since they could have been implemented using `RUN_BLACK_EXTRA_ARGS`

To implement it correctly, the shell script would need to parse `RUN_BLACK_EXTRA_ARGS` to check if they include `--include ...`. Because if it is missing, the script would need to add `--include "^[^.].*\.py$+ä+*"` to make it the default. For `--ignore` for flake8, the default would be not to include a a command-line flag since the defaults are in tox.ini.

> So for the purpose of setting a good coding style precedent for this and future code (as opposed to being consistent with runblack.sh), I should get rid of `RUN_FLAKE8_IGNORE` and instead allow this and other overrides to be done solely via `RUN_FLAKE8_EXTRA_ARGS`?

This is my suggestion. At least if there is a role that needs it.